### PR TITLE
Update axum to 0.6

### PR DESCRIPTION
**Pull request by Dispersia**
Just updates to use the newer middleware parts from axum 0.6

### CODE REVIEW

**Comment by Dispersia** [comment]
I just found out they implemented a new way of getting this through state that is type safe and faster, so trying to port this to that

**Comment by AzureMarker** [comment]
Sorry, for some reason I didn't see a notification for this until now! Thanks for opening a PR.

File: shaku_axum/src/inject_component.rs [hunk 1]
```diff
@@ -54,7 +53,7 @@ use std::sync::Arc;
 ///
 ///     let app = Router::new()
 ///         .route("/", get(hello))
-///         .layer(Extension(module));
+///         .with_state(module);
```
Comment (AzureMarker) [inline shaku_axum/src/inject_component.rs:56]: Now that the mechanism has changed, the description at the top of this documentation should be updated to say there must be an `impl FromRef<S> for Arc<M>`.

File: shaku_axum/src/inject_provided.rs [hunk 2]
```diff
@@ -50,7 +51,7 @@ use std::ops::Deref;
 ///
 ///     let app = Router::new()
 ///         .route("/", get(hello))
-///         .layer(Extension(module));
+///         .with_state(module);
```
Comment (AzureMarker) [inline shaku_axum/src/inject_provided.rs:54]: Same comment about `impl FromRef<S> for Arc<M>` here

**Review by AzureMarker — COMMENTED** [review]
I had to update the Rust version used since axum's min version is now 1.60.

Just left a small doc comment, otherwise looks good!

**Comment by Dispersia** [comment]
No worries! Thanks for the review. I am pretty bad at documentation, I added the lines, however it feels a bit odd. The comment is attached to the struct `Inject`, however S does not exist on it, it's a part of FromRequestParts, which is hidden from it. Is that ok, or do you know of a better way of showing that data? I can add additional `where S is Axum State`, but didn't know if that also didn't describe enough

**Comment by AzureMarker** [comment]
No worries, I'll take a look after merging. Thanks again for the PR!

**Comment by AzureMarker** [comment]
Released shaku_axum 0.4.0

**Pull request merged**